sessions/TypeScript-100/common-types-examples.ts

```diff
@@ -0,0 +1,150 @@
+const isFun: boolean = false;
+
+// there is no int or float, everything is number to JS/TS
+const pie: number = 3.14;
+
+// strings can use single or double quotes
+const fullName: string = "Bob Bobbington";
+
+// template string, using backticks
+const greet: string = `Salutations ${fullName}!`
+
+// symbol
+// note that both variables have what seems to be the same value
+const firstName = Symbol("name");
+const secondName = Symbol("name");
+// if we check their equality, it will always return false & TS will give us an error
+if (firstName === secondName) {
+ // cannot ever happen as each Symbol is a globally unique reference
+}
+
+// bigint
+// creation via the BigInt function
+const big1: bigint = BigInt(200);
+// creation via the literal syntax
+let big2: bigint = 200n;
+
+const uninitialisedValue: undefined = undefined;
+
+const absentValue: null = null;
```

Comment:
I think this is not a common use-case, would it be a better example to show how you can explicitly type something to allow null or a different type?

```suggestion
let potentiallyAbsentName: string | null = null;
// potentiallyAbsentName === null right now, but could be assigned a string value.
```